connectors/services/job_scheduling.py

```diff
@@ -88,9 +88,19 @@ async def _schedule(self, connector):
             raise DataSourceError(msg)
 
         source_klass = get_source_klass(self.source_list[connector.service_type])
+        data_source = source_klass(connector.configuration)
+        data_source.set_logger(connector.logger)
+
+        connector.log_debug("Validating configuration")
+        try:
+            data_source.validate_config_fields()
+            await data_source.validate_config()
```

Comment:
should we go ahead and call `data_source.ping` while we're at it?

Reply:
I wanted to do so, but decided to split it into 2 prs for simplicity of reviewing :D